source/darwin/darwin_pki_utils.c

```diff
@@ -104,6 +104,64 @@ int aws_import_public_and_private_keys_to_identity(
         goto done;
     }
 
+    /*
+     * If the key format is unknown, we tried to decode the key into DER format import it.
+     * The PEM file might contians multiple key sections, we will only add the first succeed key into the keychain.
+     */
+    if (key_status == errSecUnknownFormat) {
+        AWS_LOGF_TRACE(AWS_LS_IO_PKI, "static: error reading private key format, try ECC key format.");
+        struct aws_array_list decoded_key_buffer_list;
+
+        /* Init empty array list, ideally, the PEM should only has one key included. */
+        if (aws_array_list_init_dynamic(&decoded_key_buffer_list, alloc, 1, sizeof(struct aws_byte_buf))) {
+            result = AWS_OP_ERR;
+            goto done;
+        }
+
+        /* Decode PEM format file to DER format */
+        if (aws_decode_pem_to_buffer_list(alloc, private_key, &decoded_key_buffer_list)) {
+            aws_array_list_clean_up(&decoded_key_buffer_list);
+            AWS_LOGF_ERROR(AWS_LS_IO_PKI, "static: Failed to decode PEM private key to DER format.");
+            result = aws_raise_error(AWS_IO_FILE_VALIDATION_FAILURE);
+            goto done;
+        }
+        AWS_ASSERT(decoded_key_buffer_list);
+
+        // A PEM file could contains multiple PEM data section. Try importing each PEM section until find the first
+        // succed key.
+        for (size_t index = 0; index < aws_array_list_length(&decoded_key_buffer_list); index++) {
+            struct aws_byte_buf *decoded_key_buffer = NULL;
+            /* We only check the first pem section. Currently, we dont support key with multiple pem section. */
+            aws_array_list_get_at_ptr(&decoded_key_buffer_list, (void **)&decoded_key_buffer, index);
+            AWS_ASSERT(decoded_key_buffer);
+            key_data = CFDataCreate(cf_alloc, decoded_key_buffer->buffer, decoded_key_buffer->len);
+            if (!key_data) {
+                aws_byte_buf_clean_up_secure(decoded_key_buffer);
+                AWS_LOGF_ERROR(AWS_LS_IO_PKI, "static: error in creating ECC key data system call.");
+                result = aws_raise_error(AWS_ERROR_SYS_CALL_FAILURE);
```

Comment:
nit: this will pollute the errors and can confuse anyone who calls aws_last_error